test/odbc/psqlodbc/test_pg_bpchar.cpp

```diff
@@ -0,0 +1,724 @@
+  const vector<int> LENGTH_EXPECTED = {4, 1};
+  const vector<int> PRECISION_EXPECTED = {0, 0};
+  const vector<int> SCALE_EXPECTED = {0, 0};
+  const vector<string> NAME_EXPECTED = {"int4", "unknown"};
+  const vector<string> PREFIX_EXPECTED = {"int4", "'"};
+  const vector<string> SUFFIX_EXPECTED = {"int4", "'"};
+  const vector<int> IS_CASE_SENSITIVE = {0, 0};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_1);
+  testCommonCharColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_1.size(), COL1_NAME, LENGTH_EXPECTED, 
+    PRECISION_EXPECTED, SCALE_EXPECTED, NAME_EXPECTED, IS_CASE_SENSITIVE, PREFIX_EXPECTED, SUFFIX_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<int> LENGTH_EXPECTED_4000 = {4, 4000};
+  const vector<int> PRECISION_EXPECTED_4000 = {0, 0};
+  const vector<int> SCALE_EXPECTED_4000 = {0, 0};
+  const vector<string> NAME_EXPECTED_4000 = {"int4", "unknown"};
+  
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_4000);
+  testCommonCharColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_4000.size(), COL1_NAME, LENGTH_EXPECTED_4000, 
+    PRECISION_EXPECTED_4000, SCALE_EXPECTED_4000, NAME_EXPECTED_4000, IS_CASE_SENSITIVE, PREFIX_EXPECTED, SUFFIX_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<int> LENGTH_EXPECTED_20 = {4, 20};
+  const vector<int> PRECISION_EXPECTED_20 = {0, 0};
+  const vector<int> SCALE_EXPECTED_20 = {0, 0};
+  const vector<string> NAME_EXPECTED_20 = {"int4", "unknown"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20);
+  testCommonCharColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20.size(), COL1_NAME, LENGTH_EXPECTED_20, 
+    PRECISION_EXPECTED_20, SCALE_EXPECTED_20, NAME_EXPECTED_20, IS_CASE_SENSITIVE, PREFIX_EXPECTED, SUFFIX_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Bpchar, Table_Create_Fail) {
+  const vector<vector<pair<string, string>>> INVALID_COLUMNS {
+    {{"invalid1", DATATYPE + "(-1)"}},
+    {{"invalid1", DATATYPE + "(0)"}},
+    {{"invalid1", DATATYPE + "(NULL)"}}
+  };
+  testTableCreationFailure(ServerType::PSQL, TABLE_NAME, INVALID_COLUMNS);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Bpchar, Insertion_Success) {
+  const vector<string> INSERTED_VALUES_1 = {
+    "NULL", 
+    STRING_1,
+    "" 
+  };
+
+  const vector<string> EXPECTED_VALUES_1 = {
+    "NULL", 
+    STRING_1,
+    " " 
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_1);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES_1, EXPECTED_VALUES_1);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> INSERTED_VALUES_4000 = {
+    "NULL", // NULL values
+    "",
+    STRING_1,
+    STRING_4000,
+    STRING_20
+  };
+
+  const vector<string> EXPECTED_VALUES_4000 = {
+    "NULL", // NULL values
+    "" + std::string(4000, ' '),
```

Comment:
Although not checked in yet, you can take a look at the following which might help with padding the string.

Function itself
https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/2492d1547e8447adb822eaa50726850ea75957ef/test/odbc/psqlodbc/psqlodbc_tests_common.cpp#L3

Example of usage
https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/2492d1547e8447adb822eaa50726850ea75957ef/test/odbc/psqlodbc/char.cpp#L136

Reply:
std::string looks very simple, I think you suggest me to use this long ago, before I used a function to achieve this as well